utils/cronjobs_osgeo_lxd/hugo_clean_and_update_job.sh

```diff
@@ -3,31 +3,25 @@
 # 2020-2024, Markus Neteler
 # deploy updated web site from github repo
 
-# preparation
+# preparations:
 #  sudo chown -R neteler.users /var/www
+# get grass-website repo
+#  cd ~/
+#  git clone https://github.com/OSGeo/grass-website.git
+####
 
-# 0.  change into local git repo copy
-#     cd ~ ; git clone https://github.com/OSGeo/grass-website.git
-#     cd grass-website
-# 1.  update local repo from github
-# 2.  rm previously built pages in local git repo copy
-# 3.  build updated pages with hugo
-# 4.  create tmp target web directory
-# 5.  copy over updated pages to tmp target web directory
-# 6.  rm previously deployed web pages while not deleting src code files in their adjacent directory (careful!)
-# 7.  copy over updated pages from tmp target web directory to server target web directory
-# 8.  links src code dir content into now deployed web directory
-# 9.  rm tmp target web directory
-# 10. restore linked src file timestamps from their original source time stamps
+# 1. change into local git repo copy
+# 2. update local repo from github
+# 3. build updated pages with hugo into clean directory
+# 4. rsync over updated pages to target web directory, deleting leftover files
+# 5. generate links from src code directory content into web directory
+# 6. restore timestamps of links from their original time stamps in src directory
 
 cd /home/neteler/grass-website/ && \
    git pull origin master && \
    rm -rf /home/neteler/grass-website/public/* && \
    nice /home/neteler/go/bin/hugo && \
-   mkdir /var/www/html_new && \
-   \cp -rp /home/neteler/grass-website/public/* /var/www/html_new/ && \
-   rm -fr /var/www/html/* && \
-   \mv /var/www/html_new/* /var/www/html/ && \
+   rsync -a --delete /home/neteler/grass-website/public/ /var/www/html/ && \
    ln -s /var/www/code_and_data/* /var/www/html/ && \
-   rmdir /var/www/html_new && \
-   (cd /var/www/html/ ; /home/neteler/bin/fix_link_timestamp.sh .)
+   (cd /var/www/html/ ; /home/neteler/bin/fix_link_timestamp.sh .) && \
+   exit 0
```

Comment:
Any special reason for adding `exit 0`?

Reply:
It indicates that the entire `utils/cronjobs_osgeo_lxd/hugo_clean_and_update_job.sh` was successfully completed and not the last executed script within (here: `fix_link_timestamp.sh`).

Reply:
I don't see that. With `&&`, if cd and fix_link_timestamp.sh return non-zero, that's what will be returned from the script. If they return zero, zero will be returned from the script anyway. Or not? Or is this some Bash best practice? We don't end this with `|| exit 1`.

I'm just trying to avoid wondering what this is later on. We can merge whatever.

Reply:
Okay, agreed. Reverted in 768661da66b9d9e2381c6a0885285aa2cc63b013